File: ReSharper.FSharp/src/FSharp.Psi.Features/src/Injected/ZoneMarker.fs
namespace JetBrains.ReSharper.Plugins.FSharp.Psi.Features.Injected

open JetBrains.Application.BuildScript.Application.Zones
open JetBrains.Platform.RdFramework
open JetBrains.ReSharper.Feature.Services
open JetBrains.ReSharper.Psi.CSharp
open JetBrains.ReSharper.Psi.RegExp

[<ZoneMarker>]
type ZoneMarker() =
    interface IRequire<ICodeEditingZone>
    interface IRequire<ILanguageCSharpZone>
    interface IRequire<ILanguageRegExpZone>
    interface IRequire<IRdFrameworkZone>
    interface IRequire<ISinceClr4HostZone>

File: ReSharper.FSharp/src/FSharp.Psi/src/Features/Daemon/ContextHighlighters/ZoneMarker.cs
using JetBrains.Application.BuildScript.Application.Zones;
using JetBrains.Platform.RdFramework;
using JetBrains.ReSharper.Feature.Services.Daemon;
using JetBrains.ReSharper.Psi.CSharp;

namespace JetBrains.ReSharper.Plugins.FSharp.Psi.Features.Daemon.ContextHighlighters
{
  [ZoneMarker]
  public class ZoneMarker : IRequire<DaemonEngineZone>, IRequire<DaemonZone>, IRequire<ILanguageCSharpZone>, IRequire<IRdFrameworkZone>, IRequire<ISinceClr4HostZone>
  {
  }
}